thriftpy2/protocol/apache_json.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+"""
+Transport for json protocol that apache thrift files will understand
+unfortunately, thriftpy2's TJSONProtocol is not compatible with apache's
+"""
+from __future__ import absolute_import
+import json
+import base64
+
+from six import string_types
+from thriftpy2.protocol import TProtocolBase
+from thriftpy2.thrift import TType
+
+
+CTYPES = {
+    TType.BOOL: 'tf',
+    TType.BYTE: 'i8',
+    TType.I16: 'i16',
+    TType.I32: 'i32',
+    TType.I64: 'i64',
+    TType.DOUBLE: 'dbl',
+    TType.STRING: 'str',
+    TType.BINARY: 'str',  # apache sends binary data as base64 encoded
+    TType.STRUCT: 'rec',
+    TType.LIST: 'lst',
+    TType.SET: 'set',
+    TType.MAP: 'map',
+}
+
+JTYPES = {v: k for k, v in CTYPES.items()}
+
+VERSION = 1
+
+
+def flatten(suitable_for_isinstance):
+    """
+    isinstance() can accept a bunch of really annoying different types:
+        * a single type
+        * a tuple of types
+        * an arbitrary nested tree of tuples
+    Return a flattened tuple of the given argument.
+    """
+
+    types = list()
+
+    if not isinstance(suitable_for_isinstance, tuple):
+        suitable_for_isinstance = (suitable_for_isinstance,)
+    for thing in suitable_for_isinstance:
+        if isinstance(thing, tuple):
+            types.extend(flatten(thing))
+        else:
+            types.append(thing)
+    return tuple(types)
+
+
+class TApacheJSONProtocolFactory:
```

Comment:
All classes should be inherit from object for we have python2 compatible.

Reply:
Fixed ffb624c